### PR TITLE
WIP: HTTP tracing

### PR DESCRIPTION
**Pull request by ramonza**

### CODE REVIEW

**Comment by odeke-em** [comment]
@ramonza thanks for the PR, weirdly googlebot can't find your email though.

File: trace/HTTP.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Tracing HTTP
+
+## Client
+
+* Client span name format
+* Message send events
+* Message receive events
+* Status
+* Attributes names
+** Host
```
Comment (rakyll) [inline trace/HTTP.md]: We might need query parameters here as well.

File: trace/HTTP.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+** Resolved IP of connected socket
+** Path
+** Protocol version
+* Transport layer details (how do we represent TLS negotiation)
+* Redirects (how are these to be represented)
```
Comment (rakyll) [inline trace/HTTP.md]: Shouldn't each redirect represented as its own span?

File: trace/HTTP.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+* Client span name format
+* Message send events
+* Message receive events
+* Status
+* Attributes names
+** Host
+** Resolved IP of connected socket
+** Path
+** Protocol version
+* Transport layer details (how do we represent TLS negotiation)
```
Comment (rakyll) [inline trace/HTTP.md]: We can add more such as DNS and more.

File: trace/HTTP.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+* Redirects (how are these to be represented)
+* Retries (should be logged as separate requests)
+* https://blog.golang.org/http-tracing
```
Comment (rakyll) [inline trace/HTTP.md]: This might be under transport layer details.

File: trace/HTTP.md (hunk 2 above)
Comment (adriancole) [reply trace/HTTP.md]: yes

File: trace/HTTP.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Tracing HTTP
+
+## Client
+
+* Client span name format
```
Comment (adriancole) [inline trace/HTTP.md]: in zipkin we use the http method name due to fixed cardinality. For platforms that support route templates, that's the better choice. https://github.com/openzipkin/zipkin/issues/1874

**Pull request closed without merging**